source/_common.R

```diff
@@ -76,3 +78,20 @@ get_var = function(name) {
   }
   return (get(name))
 }
+
+# See: https://stackoverflow.com/questions/76732162/is-there-a-way-to-detect-the-chunk-language-when-setting-knitr-chunk-options
+set_lang_default <- function(options, field) {
+    if (is.na(options[[field]])) {
+      options[[field]] <- ((options$engine == "python" & is_py_ed) |
```

Comment:
This is a bit confusing to be setting the value with the output of the conditional. How does a reader reason about the meaning of this function? Perhaps worth a docstring.

Perhaps make it a bit more explicit?

```
set_default_options_for_lang('python', view, TRUE, options)
```

Or similar

Reply:
Sorry - could you explain what you mean by `set_default_options_for_lang('python', view, TRUE, options)`?   Do you mean, we would have to duplicate getting the `options$engine` and more in the `opts_hooks` set step?

Reply:
I just meant it's quite confusing right now that the value is set to the logical condition. You can imagine a language default for any setting being either true or false.

Maybe "set_option_true_for_lang" captures the semantics. Or...?